lib/spack/spack/ci/__init__.py

```diff
@@ -581,20 +582,33 @@ def copy_stage_logs_to_artifacts(job_spec: spack.spec.Spec, job_log_dir: str) ->
     tty.debug(f"job spec: {job_spec}")
 
     try:
-        pkg_cls = spack.repo.PATH.get_pkg_class(job_spec.name)
-        job_pkg = pkg_cls(job_spec)
-        tty.debug(f"job package: {job_pkg}")
+        package_metadata_root = pathlib.Path(spack.store.STORE.layout.metadata_path(job_spec))
     except AssertionError:
-        msg = f"Cannot copy stage logs: job spec ({job_spec}) must be concrete"
+        msg = f"Cannot copy logs: job spec ({job_spec}) must be concrete"
         tty.error(msg)
         return
 
-    stage_dir = job_pkg.stage.path
-    tty.debug(f"stage dir: {stage_dir}")
+    # Get the package's archived files
+    archive_files = []
+    try:
+        archive_root = package_metadata_root / "archived-files"
+        if archive_root.exists():
+            assert archive_root.is_dir()
+            archive_files = [f for f in archive_root.rglob("*") if f.is_file()]
+    except AssertionError:
+        msg = "Cannot copy package archived files: archived-files must be a directory"
+        tty.error(msg)
+        return
+
+    build_log_zipped = package_metadata_root / "spack-build-out.txt.gz"
+    assert build_log_zipped.is_file()
```

Comment:
Just reading this quickly, but it looks like you want the job to fail if there are no logs?

Reply:
ah, bleh, yeah I see we just warn for those in the copy command. I will drop these.